code/modules/power/cable.dm

```diff
@@ -165,23 +165,31 @@ var/list/possible_cable_coil_colours = list(
 			to_chat(user, "Not enough cable")
 			return
 		if(user.a_intent == I_HURT)
+			if(used_now)
+				to_chat(user, SPAN_WARNING("You are already splicing the [src.name]!")) //don't want people stacking splices on one turf
+				return
+			used_now = TRUE
 			if(locate(/obj/structure/wire_splicing) in T)
 				to_chat(user, SPAN_WARNING("There is splicing already!"))
+				used_now = FALSE
 				return
 			to_chat(user, SPAN_NOTICE("You started messsing with wires..."))
 			if(shock(user, 100)) //check if he got his insulation gloves
+				used_now = FALSE
 				return 		//he didn't
 			if(do_after(user, 20))
```

Comment:
`if(do_after(user, 20, src))` there as well

And get rid of those used_now, please `do_after` would do it for you

Reply:
I'm very new to this, so the feedback is appreciated

Reply:
removing all of the `used_now` disables the spam protection I was putting in. am I missing something?

Reply:
Well, I'm not really sure right now. Just tested it on my machine and got spam as well

Reply:
Ooooooooh, I'm fucking dumb and mistaken `tool.use_tool` with `do_after`

Reply:
So **you are very damn right all about flags** but `do_after` still needs proper calls